src/components/Tables/Table.tsx

```diff
@@ -0,0 +1,80 @@
+"use client";
+
+import React, { useState, useMemo, useEffect } from "react";
+import DataTable, { TableColumn } from "react-data-table-component";
+import TableFilterBar from "./TableFilterBar";
+
+interface Props {
+    data: { [key: string]: string | number | null }[];
+    headers: { [key: string]: string | null };
+}
+
+const Table: React.FC<Props> = (props) => {
+    const [filterText, setFilterText] = useState<{ [key: string]: string }>({});
+
+    // temporary solution to fix hydration error
+    const [domLoaded, setDomLoaded] = useState(false);
+    useEffect(() => {
+        setDomLoaded(true);
+    }, []);
+
+    const subHeaderComponentMemo = useMemo(() => {
+        const handleClear: () => void = () => {
+            if (filterText) {
+                setFilterText({});
+            }
+        };
+        return (
+            <TableFilterBar
+                onFilter={(e: React.ChangeEvent<HTMLInputElement>, filterField: string): void => {
+                    const filterTextCopy = { ...filterText };
+                    filterTextCopy[filterField] = e.target.value;
+                    setFilterText(filterTextCopy);
+                }}
+                onClear={handleClear}
+                filterText={filterText}
+                headers={props.headers}
+            />
+        );
+    }, [filterText, props.headers]);
+
+    const filteredItems = props.data.filter((item) => {
+        for (const key of Object.keys(props.headers)) {
+            if (
+                !(item[key] ?? "")
+                    .toString()
+                    .toLowerCase()
+                    .includes((filterText[key] ?? "").toLowerCase())
+            ) {
+                return false;
+            }
+        }
+        return true;
+    });
+
+    const columns: TableColumn<any>[] = Object.entries(props.headers).map(([key, value]) => {
+        return {
+            name: value,
+            selector: (row: any) => row[key],
```

Comment:
can this `any` be more specific - potentially something like 'an object which has the `key` variable as an attribute'?

Reply:
resolved as suggested